src/box/box.cc

```diff
@@ -1006,6 +1006,21 @@ bad_uri:;
 	return -1;
 }
 
+/** Check validity of a uri passed in configuration option. */
+static int
+box_check_uri(struct uri *uri, const char *option_name)
+{
+	const char *source = cfg_gets(option_name);
+	if (source == NULL)
+		return 0;
```

Comment:
Here `uri` remains potentially uninitialized if the caller didn't do it. Maybe better nullify it before return.

Reply:
Good point. Done.